go/pkg/ddlog/ddlog.go

```diff
@@ -302,6 +314,19 @@ func (p *Program) CommitTransaction() error {
 	return nil
 }
 
+// CommitTransactionChangesAsArray commits a transaction. It uses a different implementation from
+// CommitTransaction, which may yield better performance when many output records are
+// generated. Unlike with CommitTransaction, DDlog will not generate one callback for each output
+// record, but will return an array of output records (with polarity). Note that we still generate
+// one OutRecordHandler callback for each output record.
```

Comment:
for the record, I tried batching callbacks as this level as well and I didn't observe any improvement